### PR TITLE
Add timing support

### PR DESCRIPTION
**Pull request by TisButMe**
This is more a RFC than a PR, things missing:

* Tests
* time_left command support

I'm curious to know if you have any feedback on this code, and how you would test it. I was thinking trying to gen_move after submitting different scenarios, and check that the time used is the one we expect, but that would slow down the tests (since we have to wait). Any better ideas ?

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [package]
 
 name = "iomrascálaí"
-version = "0.1.0"
+version = "0.2.0"
```
Comment (ujh) [inline Cargo.toml]: We can do that, but could you please then adjust the cards on Trello? I was thinking of having 0.1.0 as our first real release when the time keeping stuff is implemented and the bot is running successfully on CGOS. But we can also make this 0.2.0 and adjust everything on Trello.

**Comment by ujh** [comment]
Well, passing tests would be a start. :stuck_out_tongue: 

But in all honesty, how about moving the code to generate the time for the next move out of `Game` into a separate struct? This would probably simplify testing it. It may or may not require you to define a trait with methods from `Game` or `Board` so that you can define an implementation for your tests.

File: Cargo.toml (hunk 1 above)
Comment (ujh) [inline Cargo.toml]: And apart from which version to use, I'd like to have a separate pull request for this. We also need to change the version in `version.rs`, add a tag, etc.

**Comment by TisButMe** [comment]
... I used to store time in s, and forgot to update tests when started using ms. All is well (all is better, anyway) now.

File: Cargo.toml (hunk 1 above)
Comment (TisButMe) [inline Cargo.toml]: I'm fine with bumping it down to 0.1.x, but I thought time-keeping was a major feature to add. I'm not quite sure of to do semver well, though, so if you disagree I'm fine with it as well.

**Comment by TisButMe** [comment]
>But in all honesty, how about moving the code to generate the time for the next move out of Game into a separate struct? This would probably simplify testing it. It may or may not require you to define a trait with methods from Game or Board so that you can define an implementation for your tests.

Agreed, I wanted to do that as well, I'll work on it.

**Comment by TisButMe** [comment]
Anyway, do you have ideas on how to test this code reasonably well, while not slowing down our testing suite down a lot ?

**Comment by ujh** [comment]
For now I'd be fine with moving the code out of `Game` and into a separate struct and testing the calculations, i.e. how much time to allocate per move. That may or may not require you to define a trait for a few functions of `Game` so that you can test your new struct in isolation.

Some time in the future I envision a system where the engine runs in a separate task and the timing code just tells the engine to start the calculation, sleeps for the time allocated for the move and then send the engine a signal for it to return its best move. But that is something for a later version, I think.

**Comment by ujh** [comment]
Is there anything I can help you with, here? I'd love to get the timing stuff merged so that we can start testing on CGOS.

**Comment by TisButMe** [comment]
We have a big deadline incoming at work, so I don't have a lot of time, but feel free to pull my branch and continue the work.

**Comment by ujh** [comment]
Thanks, that's all I wanted to know. I'll start working on it next week then. And good luck with the deadline!

**Comment by ujh** [comment]
I've create a branch called `add_timer_support` in this repository and I'll close this pull request for now. Let me know when you have time to continue working in this project!

**Pull request closed without merging**